### PR TITLE
faster build by reduce by-product, reduce linking library and fix com…

### PR DESCRIPTION
**Pull request by zhwesky2010**
1. Speed up the compilation, remove the Useless intermediate process of paddle_fluid/paddle_fluid_origin/paddle_fluid_shared/memory/paddle_framwork , and change them to be library which does't cost time. As a result, the dependency relationship between targets has been adjusted, too.

2. Reduced some target_link_libraries of testing executables. Currently, there are unnecessary libs are linked into testing executables.

3. Eliminates compilation warnings caused by std=c++11.

4. Some message warnings of cmake have been removed to avoid disturbing users.

5. Add tensorrt information to `version.txt` of inference lib.

It has been checked by **PR_Windows_CI_Inference**
http://ci.paddlepaddle.org/viewLog.html?tab=buildLog&buildTypeId=PaddleWindows_PrWindowsCiInference_2&buildId=270224&_focus=1443#_state=1442

### CODE REVIEW

File: cmake/external/protobuf.cmake [hunk 1]
```diff
@@ -138,19 +138,16 @@ IF (WIN32)
...
     find_library(PROTOBUF_LITE_LIBRARY protobuf-lite libprotobuf-lite.lib PATHS ${PROTOBUF_ROOT}/lib NO_DEFAULT_PATH)
     find_library(PROTOBUF_PROTOC_LIBRARY protoc libprotoc.lib PATHS ${PROTOBUF_ROOT}/lib NO_DEFAULT_PATH)
     find_program(PROTOBUF_PROTOC_EXECUTABLE protoc PATHS ${PROTOBUF_ROOT}/bin NO_DEFAULT_PATH)
     if (PROTOBUF_INCLUDE_DIR AND PROTOBUF_LIBRARY AND PROTOBUF_LITE_LIBRARY AND PROTOBUF_PROTOC_LIBRARY AND PROTOBUF_PROTOC_EXECUTABLE)
-        message(STATUS "Using custom protobuf library in ${PROTOBUF_ROOT}.")
         SET(PROTOBUF_FOUND true)
         SET_PROTOBUF_VERSION()
         PROMPT_PROTOBUF_LIB()
-    else()
-        message(WARNING "Cannot find protobuf library in ${PROTOBUF_ROOT}")
```
Comment (liupluswei) [inline cmake/external/protobuf.cmake:153]: revert this change

File: paddle/fluid/inference/CMakeLists.txt [hunk 2]
```diff
@@ -50,47 +52,35 @@ if(NOT ON_INFER)
...
-              DEPS ${fluid_modules} paddle_fluid_api reset_tensor_array
-                  analysis_config ${mkldnn_quantizer_cfg} paddle_pass_builder)
-else(WIN32)
-  cc_library(paddle_fluid_shared SHARED SRCS ${SHARED_INFERENCE_SRCS}
-            DEPS ${fluid_modules} paddle_fluid_api reset_tensor_array
-                  analysis_config ${mkldnn_quantizer_cfg} paddle_pass_builder)
-endif()
+# Create shared inference library defaultly
+cc_library(paddle_fluid_shared SHARED SRCS ${SHARED_INFERENCE_SRCS}
+      DEPS ${fluid_modules} paddle_fluid_api analysis_predictor)
```
Comment (liupluswei) [inline paddle/fluid/inference/CMakeLists.txt]: check paddle_fluid_api is necessary or not

File: cmake/tensorrt.cmake [hunk 3]
```diff
@@ -40,15 +37,15 @@ find_library(TENSORRT_LIBRARY NAMES ${TR_INFER_LIB} ${TR_INFER_RT}
...
-    set(TENSORRT_FOUND ON)
-  endif(WITH_DSO)
+    if(WITH_DSO)
+        set(TENSORRT_FOUND ON)
+    endif(WITH_DSO)
 else()
     set(TENSORRT_FOUND OFF)
     if(WITH_DSO)
-        message(WARNING "TensorRT is NOT found.")
+        message(STATUS "TensorRT is NOT found.")
```
Comment (liupluswei) [inline cmake/tensorrt.cmake]: revert

File: paddle/fluid/inference/CMakeLists.txt [hunk 4]
```diff
@@ -1,13 +1,26 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
 if(WITH_TESTING)
   include(tests/test.cmake) # some generic cmake funtion for inference
 endif()
 
-set(FLUID_CORE_MODULES proto_desc memory lod_tensor executor data_feed_proto)
-
```
Comment (liupluswei) [inline paddle/fluid/inference/CMakeLists.txt:6]: pay attention to inference/tensorrt/convert/CMakelists.txt

File: paddle/fluid/inference/CMakeLists.txt [hunk 5]
```diff
@@ -23,17 +36,6 @@ endif()
 
 get_property(fluid_modules GLOBAL PROPERTY FLUID_MODULES)
```
Comment (liupluswei) [inline paddle/fluid/inference/CMakeLists.txt:42]: please explain why add_subdirectory(api) is in the below

File: paddle/fluid/inference/api/CMakeLists.txt [hunk 6]
```diff
@@ -18,8 +18,7 @@ if(APPLE)
 endif(APPLE)
 
 set(inference_deps ${analysis_deps}
-  paddle_inference_api paddle_fluid_api
-  analysis pass naive_executor
+  paddle_inference_api analysis naive_executor
```
Comment (liupluswei) [inline paddle/fluid/inference/api/CMakeLists.txt]: (suggestion) dont' use string that has not been defined yet

File: paddle/fluid/inference/api/high_level_api_cn.md [hunk 7]
```diff
@@ -79,7 +78,7 @@ CHECK(predictor->Run(slots, &outputs));
 // 获取 outputs ...
 ```
 
-编译时，联编 `libpaddle_fluid.a/.so` 和 `libpaddle_inference_api.a/.so` 便可。 
+编译时，联编 `libpaddle_fluid.a/.so` 或 `libpaddle_fluid.lib/paddle_fluid.dll` 便可。
```
Comment (liupluswei) [inline paddle/fluid/inference/api/high_level_api_cn.md]: add system description

File: cmake/external/protobuf.cmake (hunk 1 above)
Comment (zhwesky2010) [reply cmake/external/protobuf.cmake:153]: Done

File: paddle/fluid/inference/CMakeLists.txt (hunk 4 above)
Comment (zhwesky2010) [reply paddle/fluid/inference/CMakeLists.txt:6]: Done

File: paddle/fluid/inference/CMakeLists.txt (hunk 5 above)
Comment (zhwesky2010) [reply paddle/fluid/inference/CMakeLists.txt:42]: Done

File: paddle/fluid/inference/api/CMakeLists.txt (hunk 6 above)
Comment (zhwesky2010) [reply paddle/fluid/inference/api/CMakeLists.txt]: Done

**Review by liupluswei — APPROVED** [review]
LGTM

File: paddle/fluid/inference/api/high_level_api_cn.md (hunk 7 above)
Comment (zhwesky2010) [reply paddle/fluid/inference/api/high_level_api_cn.md]: Done

File: paddle/fluid/inference/CMakeLists.txt (hunk 2 above)
Comment (zhwesky2010) [reply paddle/fluid/inference/CMakeLists.txt]: It is not necessary, has been done.

File: cmake/tensorrt.cmake (hunk 3 above)
Comment (zhwesky2010) [reply cmake/tensorrt.cmake]: Done
